js/client-mainmenu.js

```diff
@@ -379,11 +379,18 @@
 			var $searchForm = $('.mainmenu button.big').closest('form');
 			var $formatButton = $searchForm.find('button[name=format]');
 			var $teamButton = $searchForm.find('button[name=team]');
-			if (this.searching) {
+			if (this.searching && (!$.isArray(this.searching) || this.searching.length)) {
```

Comment:
That would break backwards compatibility with the current protocol:
- Currently searching randbats: `{searching: {format: 'randombattle'}}`
- Currently not searching: `{searching: false}`

The clearer way to write the condition is invert the if/else blocks:

```
if (!this.searching || $.isArray(this.searching) && !this.searching.length) {
// do stuff
} else {
// do different stuff
}
```

But I didn't want to add unneeded deltas.